### PR TITLE
graph parser: fix suite state polling regex

### LINKED ISSUE #2336: Auto suite-state polling task name matching
The proper task name regex is not being used to identify auto suite-polling tasks in the graph.  See `REC_SUITE_STATE` in `graph_parser.py`.

(A user tried to this: `graph = polling-task<remote-suite::remote-task>` and it did not generate a polling task because of the `-` in the local task name.)

### PR DESCRIPTION
**Pull request by matthewrmshin**
Upstream task names are not following the latest convention.

Fix #2336.

### CODE REVIEW

File: lib/cylc/graph_parser.py [hunk 1]
```diff
@@ -123,7 +123,8 @@ class GraphParser(object):
     REC_PARAMETERS = re.compile(r'<[\w,=\-+]+>')
 
     # Detect and extract suite state polling task info.
-    REC_SUITE_STATE = re.compile('(\w+)(<([\w\.\-/]+)::(\w+)(:\w+)?>)')
+    REC_SUITE_STATE = re.compile(
+        r'(\w+)(<([\w\.\-/]+)::(\w[\w\-+%@]*)(:\w+)?>)')
```
Comment (hjoliver) [inline lib/cylc/graph_parser.py]: You've fixed the regex for the remote task name, but not the one for the local polling task name.

Comment (hjoliver) [inline lib/cylc/graph_parser.py]: (And can we import and use the source regexes for suite and task names, in case of future changes to those?)

Comment (matthewrmshin) [inline lib/cylc/graph_parser.py]: I suppose so, and I suppose we should do the ones above as well.

**Comment by dvalters** [comment]
On reflection, perhaps there could be a couple more unit tests added for various forms that should be valid and parse correctly?

File: lib/cylc/graph_parser.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/cylc/graph_parser.py]: Done.

**Comment by matthewrmshin** [comment]
Change should now be sufficient.

**Comment by matthewrmshin** [comment]
(Note: we still have an outstanding issue for suite name #2281.)

**Pull request merged**